pylabnet/hardware/nanopositioners/smaract.py

```diff
@@ -0,0 +1,384 @@
+                if result_mode:
+                    self.log.warn(
+                        f'Failed to set mode to DC for positioner {self.dev_name},'
+                        f' channel {channel}'
+                    )
+            else:
+                result_mode = self._nanopositionersdll.SA_CTL_SetProperty_i32(
+                    self.dhandle, channel, self.PKEY_MOVE_MODE, self.MOVE_MODE_STEP
+                )
+                if result_mode:
+                    self.log.warn(
+                        f'Failed to set mode to step for positioner {self.dev_name},'
+                        f' channel {channel}'
+                    )
+
+        # Set frequency
+        if frequency is not None:
+
+            # Check for reasonable range
+            if 1 <= frequency <= 20000:
+                result_freq = self._nanopositionersdll.SA_CTL_SetProperty_i32(
+                    self.dhandle, channel, self.PKEY_STEP_FREQ, int(frequency)
+                )
+                if result_freq:
+                    self.log.warn(
+                        f'Failed to set step frequency to {frequency} for positioner '
+                        f'{self.dev_name}, channel {channel}'
+                    )
+
+            # Handle out of range request
+            else:
+                self.log.warn('Warning, can only set frequency within 1 Hz and 20 kHz')
+
+        # Set amplitude
+        if amplitude is not None:
+
+            # Check for reasonable range
+            bit_amp = value_to_bitval(amplitude, bits=16, min=0, max=100)
+            if 1 <= bit_amp <= 65535:
+                result_amp = self._nanopositionersdll.SA_CTL_SetProperty_i32(
+                    self.dhandle, channel, self.PKEY_STEP_AMP, bit_amp
+                )
+                if result_amp:
+                    self.log.warn(
+                        f'Failed to set step amplitude to {amplitude} for positioner '
+                        f'{self.dev_name}, channel {channel}'
+                    )
+            else:
+                self.log.warn('Warning, can only set amplitude in the range of 0 to 100 V')
+
+        # Set DC velocity
+        if dc_vel is not None:
+
+            # Check for reasonable range
+            bit_vel = int(dc_vel * (self.SCALE/100))
+            if 1 <= bit_vel <= self.SCALE*10**6:
+                result_vel = self._nanopositionersdll.SA_CTL_SetProperty_i64(
+                    self.dhandle, channel, self.PKEY_DC_VEL, bit_vel
+                )
+                if result_vel:
+                    self.log.warn(
+                        f'Failed to set DC velocity to {dc_vel} V/s for positioner '
+                        f'{self.dev_name}, channel {channel}'
+                    )
+            else:
+                self.log.warn('Warning, can only set velocity in the range of '
+                              f'{self.DC_VEL_MIN} to {self.DC_VEL_MAX} V/s')
+
+    def step(self, channel):
+        """ Takes a single step forward
```

Comment:
Why do you first define a single step? wouldn't be enough to have step_number as a parameter already here? I assume both gives same result eventually